src/highlevel_api.jl

```diff
@@ -71,7 +157,7 @@ end
 """
 Prints the system topology as a tree.
 """
-topology() = print_topology(gettopology())
+topology(htopo=nothing) = print_topology(gettopology(htopo))
```

Comment:
Similar to my other comment, should we introduce an `io` keyword here to give the user the options to opt-in/opt-out of showing IO devices?

(Also, you've added the `minimal` option to `print_topology`. As users aren't really expected to call this function directly, the option isn't really user-facing. Is this intentional?)

Reply:
RE the I/O kwarg: I like the idea, but indead I'm passing the `get_io` kwarg through `gettopology` (`topology_init` already has this). So someone sees IO devices by default:
```julia
julia> topo = gettopology(;reload=true)
Hwloc.Object: Machine

julia> topology(topo)
Machine (31.05 GB)
    Package L#0 P#0 (31.05 GB)
        NUMANode (31.05 GB)
        L3 (12.0 MB)
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#0 P#0 
                PU L#0 P#0 
                PU L#1 P#4 
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#1 P#1 
                PU L#2 P#1 
                PU L#3 P#5 
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#2 P#2 
                PU L#4 P#2 
                PU L#5 P#6 
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#3 P#3 
                PU L#6 P#3 
                PU L#7 P#7 
    HostBridge 
        PCI 00:02.0 (VGA)
            GPU "renderD128"
            GPU "card0"
        PCIBridge 
            PCI 01:00.0 (NVMExp)
                Block(Disk) "nvme0n1"
        PCIBridge 
            PCI 72:00.0 (Network)
                Net "wlp114s0"
        PCIBridge 
            PCI 73:00.0 (Other)
                Block "mmcblk0"
```
or they can set `get_io=false` like so:
```julia
julia> topo = gettopology(;reload=true, get_io=false)
Hwloc.Object: Machine

julia> topology(topo)
Machine (31.05 GB)
    Package L#0 P#0 (31.05 GB)
        NUMANode (31.05 GB)
        L3 (12.0 MB)
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#0 P#0 
                PU L#0 P#0 
                PU L#1 P#4 
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#1 P#1 
                PU L#2 P#1 
                PU L#3 P#5 
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#2 P#2 
                PU L#4 P#2 
                PU L#5 P#6 
            L2 (1.25 MB) + L1 (48.0 kB) + Core L#3 P#3 
                PU L#6 P#3 
                PU L#7 P#7 
```

How does that sound?

The minimal classes is a bit of a hack -- it's mainly used the squash the absolutely obscene number of I/O devices on semi-modern apple computers, and clusters. I can remove it if it causes confusion. So `minimal` is only used to filter out the noise and make `print_topology` more like `lstopo`. I've documentat this behaviour in the docstring now:
```julia
julia> print_topology()

Machine (31.05 GB)
    Package L#0 P#0 (31.05 GB)
        NUMANode (31.05 GB)
        L3 (12.0 MB)
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#0 P#0 
                PU L#0 P#0 
                PU L#1 P#4 
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#1 P#1 
                PU L#2 P#1 
                PU L#3 P#5 
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#2 P#2 
                PU L#4 P#2 
                PU L#5 P#6 
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#3 P#3 
                PU L#6 P#3 
                PU L#7 P#7 
    HostBridge 
        PCI 00:02.0 (VGA)
            GPU "renderD128"
            GPU "card0"
        PCIBridge 
            PCI 01:00.0 (NVMExp)
                Block(Disk) "nvme0n1"
        PCIBridge 
            PCI 72:00.0 (Network)
                Net "wlp114s0"
        PCIBridge 
            PCI 73:00.0 (Other)
                Block "mmcblk0"

julia> print_topology(;minimal=false)

Machine (31.05 GB)
    Package L#0 P#0 (31.05 GB)
        NUMANode (31.05 GB)
        L3 (12.0 MB)
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#0 P#0 
                PU L#0 P#0 
                PU L#1 P#4 
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#1 P#1 
                PU L#2 P#1 
                PU L#3 P#5 
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#2 P#2 
                PU L#4 P#2 
                PU L#5 P#6 
            L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#3 P#3 
                PU L#6 P#3 
                PU L#7 P#7 
    HostBridge 
        PCI 00:00.0 (HostBridge)
        PCI 00:02.0 (VGA)
            GPU "renderD128"
            GPU "card0"
        PCI 00:04.0 (SignalProcessing)
        PCIBridge 
            PCI 01:00.0 (NVMExp)
                Block(Disk) "nvme0n1"
        PCI 00:0a.0 (SignalProcessing)
        PCI 00:0d.0 (USB)
        PCI 00:0d.2 (USB)
        PCI 00:0d.3 (USB)
        PCI 00:12.0 (Serial)
        PCI 00:14.0 (USB)
        PCI 00:14.2 (RAM)
        PCI 00:15.0 (SerialBus)
            DMA "dma0chan0"
            DMA "dma0chan1"
        PCI 00:15.1 (SerialBus)
            DMA "dma1chan0"
            DMA "dma1chan1"
        PCI 00:16.0 (Communication)
        PCI 00:19.0 (SerialBus)
            DMA "dma2chan0"
            DMA "dma2chan1"
        PCI 00:19.1 (SerialBus)
            DMA "dma3chan0"
            DMA "dma3chan1"
        PCIBridge 
            PCI 72:00.0 (Network)
                Net "wlp114s0"
        PCIBridge 
            PCI 73:00.0 (Other)
                Block "mmcblk0"
        PCI 00:1f.0 (ISABridge)
        PCI 00:1f.3 (MultimediaAudio)
        PCI 00:1f.4 (SMBus)
        PCI 00:1f.5 (SerialBus)

Reply:
Ok, thanks. I'm fine with `minimal` and `get_io=false` (although just `io=false` might be nicer, but whatever I guess).

Just one more thing: I note that the regular (i.e. non-io) printing has changed. Specifically, there seem to be new line breaks for the lines starting with `+`. That is, we now get

```
L2 (1.25 MB)
             + L1 (48.0 kB)
             + Core L#0 P#0 
                PU L#0 P#0 
                PU L#1 P#4
```
instead of the previous, more compact
```
L2 (256.0 kB) + L1 (32.0 kB) + Core L#0 P#0 
                PU L#0 P#0 
                PU L#1 P#1
```
I prefer the old behavior.

Reply:
oops! fixed it. Also changed all references of `get_io` to just `io`